### PR TITLE
Add support for attachment of inline volumes migrated to CSI

### PR DESCRIPTION
**Pull request by ddebroy**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
This PR adds support in the external-attacher to handle volume attachments associated with inline volumes that do not have a regular PV associated with it.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
Related to https://github.com/kubernetes/kubernetes/pull/77703

**Does this PR introduce a user-facing change?**:
```release-note
Support attachment of inline volumes migrated to CSI
```

### CODE REVIEW

File: pkg/controller/csi_handler_test.go [hunk 1]
```diff
@@ -888,12 +1049,12 @@ func TestCSIHandler(t *testing.T) {
 		{
 			name:           "detach invalid PV reference-> error",
 			initialObjects: []runtime.Object{pvWithFinalizer(), node()},
-			addedVA:        deleted(vaWithNoPVReference(va(true, fin, ann))),
+			addedVA:        deleted(vaWithNoPVReferenceNorInlineVolumeSpec(va(true, fin, ann))),
```
Comment (davidz627) [inline pkg/controller/csi_handler_test.go]: I think we're missing a test where both `PV` and `InlineVolumeSpec` are set. Hate to make you write another test after already writing this massive amount, thanks for making sure this is all well tested!

**Comment by jsafrane** [comment]
/approve
Good work! Especially thanks for keeping unit test coverage high!

File: pkg/controller/csi_handler_test.go (hunk 1 above)
Comment (jsafrane) [reply pkg/controller/csi_handler_test.go]: Good catch, we should have an unit test to cover this case. Otherwise I can't find anything wrong in this PR.

**Comment by ddebroy** [comment]
Thanks for reviewing! Comments addressed.

**Comment by davidz627** [comment]
/lgtm
great work!

**Pull request merged**